networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java

```diff
@@ -128,6 +141,15 @@ public static PeerChainValidator create(
     }
   }
 
+  private boolean remoteEpochIsInvalid(
+      final UnsignedLong currentEpoch, final UnsignedLong remoteFinalizedEpoch) {
+    // Remote finalized epoch is invalid if it is from the future
+    return remoteFinalizedEpoch.compareTo(currentEpoch) > 0
+        // Remote finalized epoch is invalid if is from the current epoch (unless we're at genesis)
+        || (remoteFinalizedEpoch.compareTo(currentEpoch) == 0
+            && remoteFinalizedEpoch.compareTo(UnsignedLong.valueOf(Constants.GENESIS_EPOCH)) > 0);
```

Comment:
nit: It's probably simpler to say:
```suggestion
            && !remoteFinalizedEpoch.equals(UnsignedLong.valueOf(Constants.GENESIS_EPOCH)));
```
Possibly slightly more accurate too if `GENESIS_BLOCK` isn't actually 0.